src/editors/EditorContainer.tsx

```diff
@@ -114,6 +139,8 @@ export default function EditorContainer<R, SR>({
       e.stopPropagation();
     } else if (['Enter', 'Tab', 'ArrowUp', 'ArrowDown', 'ArrowLeft', 'ArrowRight'].includes(e.key)) {
       commit();
+    } else if (e.key === 'Escape') {
+      commitCancel();
```

Comment:
Need to call cancel changes otherwise changes are committed on `Escape`
https://github.com/adazzle/react-data-grid/blob/canary/src/masks/InteractionMasks.tsx#L180 this line closes the editor so we need to set `changeCanceled` flag